wasm-rpc/src/value_and_type.rs

```diff
@@ -569,3 +571,232 @@ impl IntoValue for crate::RpcError {
         ])
     }
 }
+
+impl From<WitType> for AnalysedType {
+    fn from(value: WitType) -> Self {
+        assert!(!value.nodes.is_empty());
+        build_tree(&value.nodes[0], &value.nodes)
+    }
+}
+
+fn build_tree(node: &WitTypeNode, nodes: &[WitTypeNode]) -> AnalysedType {
```

Comment:
Looks like this logic is repeated?

Reply:
Where?

Reply:
There is a similar one for building `WitValue`. This one is building `WitType`.